windows/src/desktop/setup/bootstrapmain.pas

```diff
@@ -84,20 +83,45 @@ implementation
   SetupForm,
   SFX,
   TntDialogHelp,
+  UfrmDownloadProgress,
   UfrmRunDesktop,
   Upload_Settings,
   utilexecute;
 
-function CheckForOldVersionScenario: Boolean; forward;
-procedure InstallKeyboardsInOldVersion(const ShellPath: string); forward;
-procedure DoExtractOnly(FSilent: Boolean; const FExtractOnly_Path: string); forward;
-function CreateTempDir: string; forward;
-procedure RemoveTempDir(const path: string); forward;
-procedure ProcessCommandLine(var FPromptForReboot, FSilent, FForceOffline, FExtractOnly, FContinueSetup, FStartAfterInstall, FDisableUpgradeFrom6Or7Or8: Boolean; var FPackages, FExtractPath, FTier: string); forward;
-procedure SetExitVal(c: Integer); forward;
-function IsKeymanDesktop7Installed: string; forward;
-function IsKeymanDesktop8Installed: string; forward;
-function GetResourcesFromOnline(FSilent: Boolean; var FForceOffline: Boolean): Boolean; forward;
+type
+  TSetupBootstrap = class
```

Comment:
In order to use the TfrmDownloadProgress dialog, we have to have a class to hold a callback function. This change just refactors all the above functions into the class with no functionality changes.